### PR TITLE
[ABW-3796] Show Using Dapps section on account locker claims

### PR DESCRIPTION
**Pull request by matiasbzurovski**
Jira ticket: [ABW-3796](https://radixdlt.atlassian.net/browse/ABW-3796)

## Description
This PR allows showing `USING DAPPS` section on Transaction Review for account locker claims.

## Video
[Demo](https://drive.google.com/file/d/1VWDZazJxXww9kMSpoq8dANUh-FDhud87/view?usp=sharing)

[ABW-3796]: https://radixdlt.atlassian.net/browse/ABW-3796?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by CyonAlexRDX — APPROVED** [review]
🥳 nice! LGTM

File: RadixWallet/Features/TransactionReviewFeature/TransactionReview+Sections.swift [hunk 1]
```diff
@@ -344,6 +350,23 @@ extension TransactionReview {
...
+	private func extractDappEntityFromLockerAddress(_ address: LockerAddress) async throws -> DappEntity? {
+		guard
+			let locker = try await onLedgerEntitiesClient.getEntity(address.asGeneral, metadataKeys: .dappMetadataKeys).locker,
+			let dAppDefinitionAddress = locker.metadata.dappDefinition
+		else {
+			return nil
+		}
+		let metadata = try await onLedgerEntitiesClient.getDappMetadata(dAppDefinitionAddress)
+		return DappEntity(id: dAppDefinitionAddress, metadata: metadata)
+	}
```
Comment (GhenadieVP) [inline RadixWallet/Features/TransactionReviewFeature/TransactionReview+Sections.swift:368]: hmm, is extracting DappEntity from Locker is different compared to other components? I would expect the current `extractDappEntity` to be used for lockers as well.

**Review by GhenadieVP — CHANGES_REQUESTED** [review]
I am not 100% sure about it but can you check if extracting dApp Entities for Lockers can't work the same way as for other components.

File: RadixWallet/Features/TransactionReviewFeature/TransactionReview+Sections.swift [hunk 2]
```diff
@@ -313,9 +316,12 @@ extension TransactionReview {
 
 	private func extractDapps(
 		componentAddresses: [ComponentAddress],
+		lockerAddresses: [LockerAddress],
```
Comment (GhenadieVP) [inline RadixWallet/Features/TransactionReviewFeature/TransactionReview+Sections.swift:319]: I think it should work if we have a single argument `addresses: [Address]`, and then `extractDappEntities` is reused.

Comment (matiasbzurovski) [reply RadixWallet/Features/TransactionReviewFeature/TransactionReview+Sections.swift]: updated

File: RadixWallet/Features/TransactionReviewFeature/TransactionReview+Sections.swift (hunk 1 above)
Comment (matiasbzurovski) [reply RadixWallet/Features/TransactionReviewFeature/TransactionReview+Sections.swift]: updated

**Review by GhenadieVP — APPROVED** [review]
Nice!

**Pull request merged**